File: easy/1133_largest_unique_number/solution.cpp
#include <unordered_map>
#include <vector>

class Solution {
public:
  int largestUniqueNumber(std::vector<int> &nums) {
    std::unordered_map<int, int> numsCounter;

    for (const int &num : nums)
      ++numsCounter[num];

    int largestNum = -1;

    for (const auto &[num, count] : numsCounter) {
      if (count == 1)
        largestNum = std::max(num, largestNum);
    }

    return largestNum;
  }
};

File: easy/1133_largest_unique_number/solution.py
from collections import Counter
from typing import List


class Solution:
    def largestUniqueNumber(self, nums: List[int]) -> int:
        largest_unique_num = -1

        for num, count in Counter(nums).items():
            if count == 1:
                largest_unique_num = max(num, largest_unique_num)

        return largest_unique_num
